File: content/snippets/python/interview_lcci/05/06_number_of_1.py
def convert_integer(A: int, B: int) -> int:
    # Work with 32-bit two's complement representation
    x = (A ^ B) & 0xFFFFFFFF
    return bin(x).count('1')


def main() -> None:
    print(convert_integer(29, 15))
    print(convert_integer(1, 5))


if __name__ == "__main__":
    main()

File: content/snippets/cpp/interview_lcci/05/06_number_of_1.cpp
#include <iostream>

// Count the number of bits you must flip to convert integer A to integer B.
int convertInteger(int A, int B) {
    unsigned int x = (unsigned int)(A ^ B);
    int count = 0;
    while (x) { count += x & 1; x >>= 1; }
    return count;
}

int main() {
    std::cout << convertInteger(29, 15) << '\n';  // 29=11101, 15=01111 -> 2 bits differ
    std::cout << convertInteger(1, 5)  << '\n';  //  1=001, 5=101   -> 1 bit differs
}
